msal/public_client_application.go

```diff
@@ -14,24 +18,32 @@ type PublicClientApplication struct {
 	clientApplication *clientApplication
 }
 
-// CreatePublicClientApplication creates a PublicClientApplication instance given a client ID and authority URL.
-func CreatePublicClientApplication(clientID string, authority string) (*PublicClientApplication, error) {
-	clientApp := createClientApplication(clientID, authority)
-	pca := &PublicClientApplication{
-		clientApplication: clientApp,
-	}
-	return pca, nil
+// PublicClientApplicationOptions configures the PublicClientApplication's behavior.
+type PublicClientApplicationOptions struct {
+	// Accessor controls cache persistence.
+	// By default there is no cache persistence.
+	Accessor CacheAccessor
+
+	// Client sets the transport for making HTTP requests.
+	// Leave this as nil to use the default HTTP transport.
+	Client HTTPClient
 }
 
-//SetHTTPManager allows users to use their own implementation of HTTPManager.
-func (pca *PublicClientApplication) SetHTTPManager(httpManager HTTPManager) {
-	webRequestManager := createWebRequestManager(httpManager)
-	pca.clientApplication.webRequestManager = webRequestManager
+// DefaultPublicClientApplicationOptions returns an instance of PublicClientApplicationOptions initialized with default values.
+func DefaultPublicClientApplicationOptions() PublicClientApplicationOptions {
+	return PublicClientApplicationOptions{
+		Client: http.DefaultClient,
```

Comment:
Would we need to tune DefaultMaxIdleConnsPerHost or MaxIdleConnsPerHost ?

Reply:
I don't believe so but am open to the possibility.

FWIW, in the Azure SDK for Go we've been using the standard library's default HTTP client without any complaints (either it's not an issue or customers are configuring it to their requirements).